itest/lnd_funding_test.go

```diff
@@ -1249,3 +1251,239 @@ func deriveFundingShim(ht *lntest.HarnessTest, carol, dave *node.HarnessNode,
+	ht.FundCoinsUnconfirmed(fundingAmt, carol)
+
+	reqListUnspent := &walletrpc.ListUnspentRequest{}
+	respListUnspent := carol.RPC.ListUnspent(reqListUnspent)
+
+	// Verify that Carol only has a single utxo in her internal wallet.
+	utxos := respListUnspent.Utxos
+	require.EqualValues(ht, 1, len(utxos))
+
+	// We will open at max. 2 channels.
+	chanSize := fundingAmt / 3
+
+	// Open a channel to dave with an unconfirmed utxo. Although this utxo
+	// is unconfirmed it can be used to open a channel because it did not
+	// originated from the sweeper subsystem.
+	update := ht.OpenChannelAssertPending(carol, dave,
+		lntest.OpenChannelParams{
+			Amt:              chanSize,
+			SpendUnconfirmed: true,
+			CommitmentType:   cType,
+		})
+	chanPoint1 := lntest.ChanPointFromPendingUpdate(update)
+
+	// Verify that both nodes know about the channel.
+	ht.AssertNumPendingOpenChannels(carol, 1)
+	ht.AssertNumPendingOpenChannels(dave, 1)
+
+	// We open another channel on the fly, funds are unconfirmed but because
+	// the tx was not created by the sweeper we can use it and open another
+	// channel. This is a common use case when opening zeroconf channels,
+	// so unconfirmed utxos originated from prior channel opening are safe
+	// to use because channel opening should not be RBFed, at least not for
+	// now.
+	update = ht.OpenChannelAssertPending(carol, dave,
+		lntest.OpenChannelParams{
+			Amt:              chanSize,
+			SpendUnconfirmed: true,
+			CommitmentType:   cType,
+		})
+
+	chanPoint2 := lntest.ChanPointFromPendingUpdate(update)
+
+	ht.AssertNumPendingOpenChannels(carol, 2)
+	ht.AssertNumPendingOpenChannels(dave, 2)
+
+	// We expect the initial funding tx to confirm and also the two
+	// unconfirmed channel openings.
+	ht.MineBlocksAndAssertNumTxes(1, 3)
+
+	// Now we create an unconfirmed utxo which originated from the sweeper
+	// subsystem and hence is not safe to use for channel openings. We do
+	// that by dave force-closing the channel. Which let's carol sweep its
+	// to_remote output which is not encumbered by any relative locktime.
+	ht.CloseChannelAssertPending(dave, chanPoint2, true)
+	// Mine the force close commitment transaction.
+	ht.MineBlocksAndAssertNumTxes(1, 1)
+
+	// We need to wait for carol initiating the sweep of the to_remote
+	// output therefore we wait until we have 2 utxos in carol's wallet.
+	// One change output from the last channel opening and the sweep of the
+	// to_remote commitment of chanPoint2.
+	err := wait.NoError(func() error {
+		respListUnspent = carol.RPC.ListUnspent(reqListUnspent)
+		utxos = respListUnspent.Utxos
+
+		if len(utxos) != 2 {
+			return fmt.Errorf("still waiting for wallet utxos")
+		}
+
+		return nil
+	}, wait.MinerMempoolTimeout)
+
+	require.NoError(ht, err)
+
+	// We filter for the unconfirmed utxo and try to open a channel with
+	// that utxo.
+	var fundingUtxo *lnrpc.Utxo
+	for _, utxo := range utxos {
+		if utxo.Confirmations == 0 {
+			fundingUtxo = utxo
+		}
+	}
```

Comment:
Good time to use [`fn.Find`](https://github.com/lightningnetwork/lnd/blob/b1175514f9854b965d4716ac015ea15387c63081/fn/slice.go#L78-L88)